jvb/src/test/kotlin/org/jitsi/videobridge/VideobridgeTest.kt

```diff
@@ -57,12 +59,17 @@ class VideobridgeTest : ShouldSpec() {
                         resp.shouldBeInstanceOf<ErrorIQ>()
                         resp.error.condition shouldBe StanzaError.Condition.service_unavailable
                     }
-                    context("and once the conference expires") {
-                        clock.elapse(2.mins)
-                        videobridge.expireConference(conf)
-                        should("start the shutdown") {
-                            verify(exactly = 1) { shutdownService.beginShutdown() }
-                        }
+                    context("When the number of participants drops below the threshold") {
+                        repeat(10) { videobridge.endpointExpired() }
+                        videobridge.shutdownState shouldBe ShutdownState.SHUTTING_DOWN
+                        // TODO: how can we verify that the shutdownService has been called when the task is scheduled
+                        // on TaskPools.SCHEDULED_POOL
+                    }
+                    context("When the graceful shutdown period expires") {
+                        clock.elapse(60.mins)
+                        // TODO: how can we verify that the state has transitioned to SHUTTING_DOWN when the task was
+                        // scheduled on TaskPools.SCHEDULED_POOL?
+                        // videobridge.shutdownState shouldBe ShutdownState.SHUTTING_DOWN
```

Comment:
Do you have a plan to resolve this TODO?  It seems like this isn't currently verifying that the shutdown actually runs.

Reply:
Not really. I don't know if there's any good way to mock scheduled tasks.

I tested manually, but of course that's no substitute.